src/iarc7_motion/task_command_handler.py

```diff
@@ -0,0 +1,238 @@
+
+    # returns last command (ground interaction, velocity, etc.)
+    # that task returned
+    def get_last_command(self):
+        return self._last_task_commands
+
+    # returns last twist sent to LLM
+    def get_last_twist(self):
+        return self._last_twist
+    
+    # public method wrapper for getting task state
+    def get_state(self):
+        return self._task_state
+
+    # main function
+    def run(self):
+        task_commands = self._get_task_command()
+        for task_command in task_commands:
+            try:
+                self._command_implementations[type(task_command)](task_command)
+            except (KeyError, TypeError) as e:
+                rospy.logerr("Task requested unimplemented command, noping: %s", type(task_command))
+                self._handle_nop_command(None)
+
+        self._last_task_commands = task_commands
+
+    # gets desired command from running task
+    def _get_task_command(self):
+        if self._task is not None:
+            try:
+                task_request = self._task.get_desired_command()
+            except Exception as e:
+                rospy.logerr('Exception getting tasks preferred velocity')
+                rospy.logerr(str(e))
+                rospy.logerr(traceback.format_exc())
+                rospy.logerr('Task Command Handler aborted task')
+                self._task = None
+                self._task_state = task_states.TaskAborted()
+                return (task_commands.NopCommand(),)
+
+            self._task_state = task_request[0]
+
+            if isinstance(self._task_state, task_states.TaskDone):
+                self._task = None
+                return task_request[1:]
+            elif isinstance(self._task_state, task_states.TaskRunning):
+                return task_request[1:]
+            else: 
+                self._task = None
+        elif self._task_canceled:
+            self._task_state = task_states.TaskCanceled()
+        else: 
+            self._task_state = task_states.TaskAborted()
+            rospy.logerr('TaskCommandHandler ran with no task running. Setting task state to aborted.')
+
+        # no action to take, return a Nop
+        return (task_commands.NopCommand(),)
+
+    """
+    Command Handlers
+
+    Types: 
+        ground interaction: these include takeoff
+        nop command: do nothing
+        velocity command: requests a velocity (x, y, z, and angular)
+        passthrough command: passthrough to throttle, pitch, and roll commands
+
+    """
+    def _handle_nop_command(self, nop_command):
+        pass
+
+    def _handle_velocity_command(self, velocity_command):
+        self._publish_twist(velocity_command.target_twist)
+
+    def _handle_passthrough_mode_command(self, passthrough_mode_command):
+        if passthrough_mode_command.enable:
+            if self._ground_interaction_task_callback is not None:
+                rospy.logerr('Task requested a passthrough action before the last was completed')
```

Comment:
last what?

Reply:
That was direct from what was in `motion_planner.py` before so I do not know.